### PR TITLE
fix: model lsq translations don't fallback to fr (resolves #2014, #2026, #2028, #2029)

### LINKED ISSUE #2014: Privacy Policy and ToS page heading are in English when site in LSQ
**Prerequisites**

- [x] I've [searched](https://github.com/accessibility-exchange/platform/issues) for existing issues

**Describe the bug**

The Privacy Policy and ToS page heading are not localized in French when the site is in LSQ. It does however work properly when the site is in FR.

**To Reproduce**
1. Go to https://accessibilityexchange.ca/lsq/about/privacy-policy
2. Notice the page heading is in English instead of French
3. Change the languages
4. Notice the other languages work properly.

**Expected behavior**

The page heading should be in French when in LSQ

**Screenshots**

Screenshot of Privacy Policy page heading in English when site is in LSQ

### LINKED ISSUE #2026: English string appear in the French version of the Resources page
### Prerequisites

- [X] I've [searched](https://github.com/accessibility-exchange/platform/issues) for existing issues

### Describe the bug

When using the French language, a user sees English strings in several of the Resource Type sections.

### To Reproduce

1. Go to https://accessibilityexchange.ca/fr/resources/collections/the-accessible-canada-act
2. Open the Secteur and Domaine d'impact types

### Expected behavior

These strings should appear in French.

### Screenshots

Screenshot 2023-11-29 at 12 03 44 PM

### Details

_No response_

### Additional context

According to Samuel, several of these strings were translated in CrowdIn but aren't appearing. He asked: "is it possible they use the other json file for this part of the website? because the 'secteur' menu hasn't updated either."

### LINKED ISSUE #2028: When a individual/org has multiple profile languages, the list is missing a space between the colon and the first language
### Prerequisites

- [X] I've [searched](https://github.com/accessibility-exchange/platform/issues) for existing issues

### Describe the bug

Formatting issue on individual and organization pages.

### To Reproduce

1. Publish an individual page in multiple languages other than English/French
2. View the page.

### Expected behavior

Proper formatting

### Screenshots

Screenshot 2023-11-30 at 9 29 42 AM

### Details

All.

### Additional context

_No response_

### LINKED ISSUE #2029: When viewing an individual page in a language other than English or French, there is no link to switch to English/French
### Prerequisites

- [X] I've [searched](https://github.com/accessibility-exchange/platform/issues) for existing issues

### Describe the bug

When viewing an individual page in a language other than English or French, there is no link to switch to English/French.

### To Reproduce

1. Visit an individual page with content in a language other than English or French.
2. Click the link to view the page in the additional language.

### Expected behavior

The "Page also available in:" list should include English or French as applicable.

### Screenshots

Screenshot 2023-11-30 at 9 34 13 AM

### Details

All.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jobara**
Resolves #2014 
Resolves #2026
Resolves #2028
Resolves #2029  

- Renames `get_written_language_for_signed_language` to `to_written_language`
- Adds `localized_route_for_locale` helper
- Adds `route_name` helper
- Makes use of the `missingKeyCallback` to fallback lsq to fr
- Fixes rendering of langage change links
- Fixes output of translatable model properties to match requested translation
- Updates/expands tests

### Prerequisites

If this PR changes PHP code or dependencies:

- [x] I've run `composer format` and fixed any code formatting issues.
- [x] I've run `composer analyze` and addressed any static analysis issues.
- [x] I've run `php artisan test` and ensured that all tests pass.
- [x] I've run `composer localize` to update localization source files and committed any changes.

If this PR changes CSS or JavaScript code or dependencies:

- [ ] I've run `npm run lint` and fixed any linting issues.
- [ ] I've run `npm run build` and ensured that CSS and JavaScript assets can be compiled.

BEGIN_COMMIT_OVERRIDE
fix: model lsq translations don't fallback to fr (resolves #2014, #2026) (#2027)

fix: rendering of language changer links (resolves #2028, #2029) (#2027)
fix: response time in other languages not saved (#2027)
fix: can't add other engagement languages (#2027)
fix: not displaying content in translation (#2027)
END_COMMIT_OVERRIDE

### CODE REVIEW

**Comment by jobara** [comment]
@greatislander in fixing the failing test I realized that I may have removed some functionality, but I'm also not sure how that part was supposed to work or if it was actually functional on the site. I wonder if you could weigh in. Basically I removed the language bits from the controllers. It looks like you could use that to pass in a query parameter to have an individual or organization public page rendered in an alternative language if one was provided. However, I don't see how a user could have specified that query parameter. Am I understanding this correctly? Have I missed something? Is this still needed?

**Comment by jobara** [comment]
> @greatislander in fixing the failing test I realized that I may have removed some functionality, but I'm also not sure how that part was supposed to work or if it was actually functional on the site. I wonder if you could weigh in. Basically I removed the language bits from the controllers. It looks like you could use that to pass in a query parameter to have an individual or organization public page rendered in an alternative language if one was provided. However, I don't see how a user could have specified that query parameter. Am I understanding this correctly? Have I missed something? Is this still needed?

I spoke with @greatislander about this. The `$language` set from the query parameter is to allow those pages to be shown in different languages, if they've been added for them by the user/org. This for example would allow an organization to localize their public page in many languages even though the site only supports a couple. Will need to bring that functionality back. The original work for this was done under #505.

While the written language supports from HasMultimodalTranslations isn't required any more, the rough in for the ASL/LSQ upload options is still requested to remain, in case we implement those features in the future (See: #455).

**Comment by jobara** [comment]
While reviewing this, @greatislander found two other bugs ( #2029, #2028 ). If possible these should be addressed alongside this one.

File: app/helpers.php [hunk 1]
```diff
@@ -2,7 +2,9 @@
 
 use App\Settings;
 use App\Settings\GeneralSettings;
+use Illuminate\Routing\Route;
 use Illuminate\Support\Arr;
+use Illuminate\Support\Facades\Route as FacadesRoute;
```
Comment (greatislander) [inline app/helpers.php]: I'd call this `RouteFacade` instead.

**Review by greatislander — CHANGES_REQUESTED** [review]

**Review by greatislander — APPROVED** [review]

**Pull request merged**